prometheus/promhttp/http.go

```diff
@@ -190,7 +190,18 @@ func HandlerForTransactional(reg prometheus.TransactionalGatherer, opts HandlerO
 		}
 		rsp.Header().Set(contentTypeHeader, string(contentType))
 
-		w, encodingHeader, err := NegotiateEncodingWriter(req, rsp, opts.DisableCompression, compressions)
+		w, encodingHeader, closeWriter, err := NegotiateEncodingWriter(req, rsp, opts.DisableCompression, compressions)
+
+		if closeWriter != nil {
+			defer func() {
```

Comment:
@bwplotka Test will fail if you comment out this defer. 
Needed to do it that way instead of calling the function directly for golangci-lint's errcheck.

Reply:
I think there is potentially better way (do `_ := ` if we don't care about this err, or pack opt.Err log in helper func), but that's already good enough, Thanks!

Reply:
Still not resolved, plus we can kill this if != nil check for close Writer I assume (let's ensure `negotiateEncodingWriter` always sets it, even if noop).

Reply:
Sounds good, I added the empty functions you suggested also if an error is returned.